test/functional/helpers.lua

```diff
@@ -447,6 +447,7 @@ end
 --    clear('-e')
 --    clear{args={'-e'}, args_rm={'-i'}, env={TERM=term}}
 function module.clear(...)
+  luv.os_unsetenv('SHELL')
```

Comment:
Not a blocker, but would it be more correct to unset SHELL in `cmake/RunTests.cmake` ?

Reply:
> It's probably less confusing to set `SHELL=sh` explicitly? (Except on Windows)

Would nvim even care about $SHELL value on Windows? `:help 'shell'` says it's $SHELL or sh, except on Windows where it's "cmd.exe" by default. And I thought that initial intention was to make nvim use the fallback value (only omitting $SHELL), not explicitly setting it to that value. May be adding a comment would help to make it less confusing?

> Not a blocker, but would it be more appropriate to make the change in `cmake/RunTests.cmake` ?

I'll try it.

Reply:
I `grep`d in test/oldtest, here are a few interesting things I've found:

$SHELL is set to sh in Makefile:
https://github.com/neovim/neovim/blob/4258f4d53cb3d8d0adbaccf410e1bb422db9badb/test/old/testdir/Makefile#L12
I don't see anything special here regarding $SHELL for Windows (and it shouldn't matter on Windows anyway I guess).

Though in setup.vim $SHELL is reset to an empty string for win32:
https://github.com/neovim/neovim/blob/4258f4d53cb3d8d0adbaccf410e1bb422db9badb/test/old/testdir/setup.vim#L70-L77

Also `shell` option is set to 'sh' directly in unix.vim:
https://github.com/neovim/neovim/blob/4258f4d53cb3d8d0adbaccf410e1bb422db9badb/test/old/testdir/unix.vim#L2-L3

So for me it still seems safer to unset it or set to an empty string.

Reply:
> I don't see anything special here regarding $SHELL for Windows (and it shouldn't matter on Windows anyway I guess).

The point is that explicitly setting SHELL to nonsense on Windows is unnecessary potential confusion so why not avoid it there? 



> So for me it still seems safer to unset it or set to an empty string.

Why? Eventhing in test/old/ is a separate set of tests. If SHELL=sh is done there, that implies setting SHELL=sh in RunTest.cmake may fix the issue, and matches the approach already taken in test/old/.

Reply:
>But then if clear{env={SHELL=foo}} is specified by a caller, that should be respected.

I don't see why it wouldn't be respected right now. I moved unsetting `$SHELL` to `cmake/RunTests.cmake` as you suggested. So parent process won't inherit `$SHELL` during test run, but if value for `$SHELL` is explicitly passed to `clear` it should just work. Or am I missing something?

Reply:
> > But then if clear{env={SHELL=foo}} is specified by a caller, that should be respected.
> 
> I don't see why it wouldn't be respected right now.

there i was referring to the case where SHELL is set in clear(). not relevant if SHELL is set in RunTests.cmake